File: Desktop/src/institution/interlink/Internship.java
package institution.interlink;

import person.Student;

import java.util.ArrayList;
import java.util.List;

public class Internship {

    private List<Student> studentList = new ArrayList<>();
    private String internshipName;

    public Internship(String name) {
        this.internshipName = name;
    }

    public void setStudent(Student student) {
        studentList.add(student);
    }

    public List<Student> getStudents() {
        return studentList;
    }
}

File: InternshipTest/Institution/InterLink/Internship.cs
using InternshipTest.Person;
using System.Text;

namespace InternshipTest.Institution.InterLink
{
    public class Internship
    {
        public Internship(string name, University university)
        {
            this.Name = name;
            this.University = university;
        }

        public string Name { get; }
        public University University { get; }

        public string GetStudents()
        {
            var builder = new StringBuilder();
            int avgLevel = this.University.GetAverageStudentsLevel();
            foreach (Student student in this.University.GetStudents())
            {
                if (student.GetStudentLevel() > avgLevel)
                {
                    builder.AppendLine(student.Name);
                }
            }

            return builder.ToString();
        }
    }
}
